src/test/java/org/jenkinsci/plugins/workflow/steps/durable_task/ShellStepTest.java

```diff
@@ -541,7 +541,7 @@ private Object writeReplace() {
         j.waitForCompletion(b);
         // Would have succeeded before https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/75.
         j.assertBuildStatus(Result.ABORTED, b);
-        j.assertLogContains("Timeout has been exceeded", b);
+        j.waitForMessage("Timeout has been exceeded", b); // TODO assertLogContains fails unless a sleep is introduced; possible race condition in waitForCompletion
```

Comment:
I saw this as well, although it passed sometimes, so a race condition seems likely.